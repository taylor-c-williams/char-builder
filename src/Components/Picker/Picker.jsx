import React from 'react';

export default function Picker({
	head,
	onHeadChange,
	middle,
	onMiddleChange,
	pants,
	onPantsChange,
	newCatchPhrase,
	setCatchPhrase,
	handleClick,
}) {
	// Set state in both app and component ??
	// const [head, setHead] = useState('dog');
	// const [middle, setMiddle] = useState('fancy');
	// const [pants, setPants] = useState('leg');

	const headPics = ['bird', 'duck', 'dog'];
	const middlePics = ['blue', 'fancy', 'pink', 'red'];
	const pantsPics = ['blue', 'dog', 'leg', 'white'];

	return (
		<section className="Picker">
			{/* Head Dropdown */}
			<label>
				Head
				<select value={head} onChange={(e) => onHeadChange(e.target.value)}>
					{headPics.map((pic) => (
						<option key={pic}>{pic}</option>
					))}
				</select>
			</label>

			{/* Middle Dropdown */}
			<label>
				Middle
				<select value={middle} onChange={(e) => onMiddleChange(e.target.value)}>
					{middlePics.map((pic) => (
						<option key={pic}>{pic}</option>
					))}
				</select>
			</label>

			{/* Pants Dropdown */}
			<label>
				Pants
				<select value={pants} onChange={(e) => onPantsChange(e.target.value)}>
					{pantsPics.map((pic) => (
						<option key={pic}>{pic}</option>
					))}
				</select>
			</label>

			{/* Catchphrase Input */}
			<label>
				Catchphrase
				<input
					defaultValue={newCatchPhrase}
					onChange={(e) => setCatchPhrase(e.target.value)}
				/>
				<button onClick={handleClick}>Add Catchphrase!</button>
			</label>
		</section>
	);
}
