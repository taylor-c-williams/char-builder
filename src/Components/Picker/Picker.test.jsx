import { screen, render } from '@testing-library/react';
import Picker from './Picker';

it('Should render Picker component', () => {
	const { container } = render(<Picker />);
	const selectHead = screen.getByLabelText(/Head:/);
	const selectMiddle = screen.getByLabelText(/Body:/);
	const selectPants = screen.getByLabelText(/Legs:/);
	const input = screen.getByLabelText(/Catchphrase:/);

	expect(selectHead).toBeInTheDocument();
	expect(selectMiddle).toBeInTheDocument();
	expect(selectPants).toBeInTheDocument();
	expect(input).toBeInTheDocument();
	expect(container).toMatchSnapshot();
});
