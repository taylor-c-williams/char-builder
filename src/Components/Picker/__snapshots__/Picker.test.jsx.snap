// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Picker component 1`] = `
<div>
  <section
    class="Picker"
  >
    <label>
      Head
      <select>
        <option>
          bird
        </option>
        <option>
          duck
        </option>
        <option>
          dog
        </option>
      </select>
    </label>
    <label>
      Middle
      <select>
        <option>
          blue
        </option>
        <option>
          fancy
        </option>
        <option>
          pink
        </option>
        <option>
          red
        </option>
      </select>
    </label>
    <label>
      Pants
      <select>
        <option>
          blue
        </option>
        <option>
          dog
        </option>
        <option>
          leg
        </option>
        <option>
          white
        </option>
      </select>
    </label>
    <label>
      Catchphrase
      <input
        id="catchphrase-button"
        type="text"
        value=""
      />
      <button>
        Add Catchphrase!
      </button>
    </label>
  </section>
</div>
`;
